features/gift_exchanges/challenge_giftexchange.feature

```diff
@@ -146,7 +146,7 @@ Feature: Gift Exchange Challenge
    Then I should see "myname4" within "#main"
      And I should see "myname3" within "#main"
      And I should see "myname2" within "#main"
-     And I should see "myname1" within "#main"
+     And I should see "myname1_pseud (myname1)" within "#main"
```

Comment:
This selector is too broad, it finds the `Request 1 by myname1_pseud (myname1)` in the request partial, so the test passes even when your bug fix is reverted. Try narrowing the within to the participant class.